### PR TITLE
Added slot statistics

### PR DESCRIPTION
**Pull request by YeahImRose**
`[p]sstats <@user>` gets the user who was mentioned
`[p]sstats` gets the information of the user who ran the command

There should be sufficient catching of potential errors, but I could've overlooked something. This can be optimized but I tried to make it readable.

### CODE REVIEW

File: cogs/economy.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import discord
 from discord.ext import commands
-from cogs.utils.dataIO import dataIO, fileIO
+from cogs.utils.dataIO import fileIO
```
Comment (tekulvw) [inline cogs/economy.py]: You  need to import `dataIO` here, not `fileIO`

File: cogs/economy.py [hunk 2]
```diff
@@ -195,13 +201,61 @@ def __init__(self, bot):
...
+        fileIO("data/economy/stats.json", "save", self.slot_stats)
+
+    def slot_stat(self, ctx, user):
+        server = ctx.message.server
+        if server.id not in self.slot_stats:
+            self.slot_stats[server.id] = {}
+            self.slot_stats[server.id][user.id] = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+        if user.id not in self.slot_stats[server.id]:
+            self.slot_stats[server.id][user.id] = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+        temp_msg = "```ruby\nSlot statistics for {0.name}:\n\nLose:\t  {1}\nx2:\t\t{2}\nx3:\t\t{3}\nx4:\t\t{4}\nx5000:\t {5}\n\n+500:\t  {6}\n+800:\t  {7}\n+1000:\t {8}\n\nSpent:\t {9}\nGained:\t{10}```".format(user, self.slot_stats[server.id][user.id][0], self.slot_stats[server.id][user.id][2], self.slot_stats[server.id][user.id][3], self.slot_stats[server.id][user.id][4], self.slot_stats[server.id][user.id][7], self.slot_stats[server.id][user.id][5], self.slot_stats[server.id][user.id][6], self.slot_stats[server.id][user.id][1], self.slot_stats[server.id][user.id][8], self.slot_stats[server.id][user.id][9])
```
Comment (tekulvw) [inline cogs/economy.py]: This thing looks like it could be done in a slightly more clean way

File: cogs/economy.py [hunk 3]
```diff
@@ -460,42 +526,42 @@ def already_in_list(self, accounts, user):
         server = ctx.message.server
         self.settings[server.id]["SLOT_MIN"] = bid
         await self.bot.say("Minimum bid is now " + str(bid) + " credits.")
-        fileIO("data/economy/settings.json", "save", self.settings)
+        dataIO.save_json(self.file_path, self.system)
```
Comment (tekulvw) [inline cogs/economy.py]: `self.system` doesn't exist

File: cogs/economy.py [hunk 4]
```diff
@@ -460,42 +526,42 @@ def already_in_list(self, accounts, user):
...
 
     @economyset.command(pass_context=True)
     async def slotmax(self, ctx, bid : int):
         """Maximum slot machine bid"""
         server = ctx.message.server
         self.settings[server.id]["SLOT_MAX"] = bid
         await self.bot.say("Maximum bid is now " + str(bid) + " credits.")
-        fileIO("data/economy/settings.json", "save", self.settings)
+        dataIO.save_json(self.file_path, self.system)
```
Comment (tekulvw) [inline cogs/economy.py]: `self.system` doesn't exist

File: cogs/economy.py [hunk 5]
```diff
@@ -460,42 +526,42 @@ def already_in_list(self, accounts, user):
...
 
     @economyset.command(pass_context=True)
     async def slottime(self, ctx, seconds : int):
         """Seconds between each slots use"""
         server = ctx.message.server
         self.settings[server.id]["SLOT_TIME"] = seconds
         await self.bot.say("Cooldown is now " + str(seconds) + " seconds.")
-        fileIO("data/economy/settings.json", "save", self.settings)
+        dataIO.save_json(self.file_path, self.system)
```
Comment (tekulvw) [inline cogs/economy.py]: `self.system` doesn't exist

File: cogs/economy.py [hunk 6]
```diff
@@ -460,42 +526,42 @@ def already_in_list(self, accounts, user):
...
 
     @economyset.command(pass_context=True)
     async def paydaytime(self, ctx, seconds : int):
         """Seconds between each payday"""
         server = ctx.message.server
         self.settings[server.id]["PAYDAY_TIME"] = seconds
         await self.bot.say("Value modified. At least " + str(seconds) + " seconds must pass between each payday.")
-        fileIO("data/economy/settings.json", "save", self.settings)
+        dataIO.save_json(self.file_path, self.system)
```
Comment (tekulvw) [inline cogs/economy.py]: `self.system` doesn't exist

File: cogs/economy.py [hunk 7]
```diff
@@ -460,42 +526,42 @@ def already_in_list(self, accounts, user):
...
 
     @economyset.command(pass_context=True)
     async def paydaycredits(self, ctx, credits : int):
         """Credits earned each payday"""
         server = ctx.message.server
         self.settings[server.id]["PAYDAY_CREDITS"] = credits
         await self.bot.say("Every payday will now give " + str(credits) + " credits.")
-        fileIO("data/economy/settings.json", "save", self.settings)
+        dataIO.save_json(self.file_path, self.system)
```
Comment (tekulvw) [inline cogs/economy.py]: `self.system` doesn't exist

**Review by tekulvw — CHANGES_REQUESTED** [review]
Have you tested this at all? You'd get at least 6 `NameError`s if you were to load it into a working bot

**Comment by Twentysix26** [comment]
Closing as this would require major rework (the code is very messy) and there's an incoming slot machine refactor

**Pull request closed without merging**